core/templates/dev/head/services/SpeechSynthesisChunkerService.js

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to chunk a piece of text into smaller parts
+ * to feed into SpeechSynthesis, because SpeechSynthesis always times
+ * out after 200-300 characters.
+ * 
+ * Code is adapted from:
+ * https://gist.github.com/woollsta/2d146f13878a301b36d7
+ * 
+ * Credits to Peter Woolley and Brett Zamir.
+ */
+
+oppia.factory('SpeechSynthesisChunkerService', [
+  '$timeout', function($timeout) {
+    var _speechSynthesis = window.speechSynthesis;
+
+    // Punctuation marks that should result in an audible pause when playing
+    // back autogenerated audio.
+    var PUNCTUATION_MARKS_TO_END_CHUNKS = '.!?:;';
+    // Punctuation marks that we don't want to influence the playing back of
+    // autogenerated audio.
+    var PUNCTUATION_MARKS_TO_IGNORE = '"';
+
+    var _speechUtteranceChunker = function(utt, settings, callback) {
```

Comment:
Perhaps add a docstring explaining types/meanings of args. E.g. I'm not sure what triggers the callback, and settings is completely vague. Possibly `settings` needs to be broken up into smaller args. It's important to fully understand this piece of code (and make it even more understandable so that we can modify it in the future if needed).

Reply:
Added docstring